src/main/java/org/prebid/server/bidder/bidmyadz/BidmyadzBidder.java

```diff
@@ -0,0 +1,120 @@
+package org.prebid.server.bidder.bidmyadz;
+
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.EncodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class BidmyadzBidder implements Bidder<BidRequest> {
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public BidmyadzBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        if (!isValidRequest(request, errors)) {
+            return Result.withErrors(errors);
+        }
+
+        try {
+            return Result.withValue(HttpRequest.<BidRequest>builder()
+                    .method(HttpMethod.POST)
+                    .uri(endpointUrl)
+                    .headers(HttpUtil.headers().add(HttpUtil.X_OPENRTB_VERSION_HEADER, "2.5"))
+                    .payload(request)
+                    .body(mapper.encode(request))
+                    .build());
+        } catch (EncodeException e) {
+            return Result.withError(
+                    BidderError.badInput(String.format("Failed to encode request body, error: %s", e.getMessage())));
+        }
+    }
+
+    private boolean isValidRequest(BidRequest request, List<BidderError> errors) {
```

Comment:
Can be static
And actually we can a bit update this method, not to check device for null
```java
  private static boolean isValidRequest(BidRequest request, List<BidderError> errors) {
        if (request.getImp().size() > 1) {
            errors.add(BidderError.badInput("Bidder does not support multi impression"));
        }

        final Device device = request.getDevice();
        final String ip = device != null ? device.getIp() : null;
        final String ipv6 = device != null ? device.getIpv6() : null;

        if (StringUtils.isEmpty(ip) && StringUtils.isEmpty(ipv6)) {
            errors.add(BidderError.badInput("IP/IPv6 is a required field"));
        }

        final String userAgent = device != null ? device.getUa() : null;

        if (StringUtils.isEmpty(userAgent)) {
            errors.add(BidderError.badInput("User-Agent is a required field"));
        }

        return errors.isEmpty();
    }
```
Then `hasValidIp()` method can be removed